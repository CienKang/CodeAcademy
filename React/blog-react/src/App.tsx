import React from 'react';
import './App.css';
import { BlogPage } from './pages';

function App() :JSX.Element {
    return (
        <div>
            <BlogPage />
        </div>
    );
}

export default App;
