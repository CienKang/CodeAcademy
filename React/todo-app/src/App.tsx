import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import './App.css';
import { LoginRegisterPage, TodoAppPage } from './pages';

function App() {
    return (
        <BrowserRouter>
            <Routes>
                <Route path='/login' element={<LoginRegisterPage />} />
                <Route path='/app' element={<TodoAppPage />} />
            </Routes>
        </BrowserRouter>
    );
}

export default App;
