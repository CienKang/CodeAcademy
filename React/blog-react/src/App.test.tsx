import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {

    it('should match with snapshots when rnedered for first time', () => {
        const {asFragment} = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});
