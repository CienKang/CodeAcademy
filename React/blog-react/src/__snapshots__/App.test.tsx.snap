// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match with snapshots when rnedered for first time 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="flex flex-col bg-black justify-center"
      >
        <h1
          class="text-white text-lg ml-32"
        >
          The Artifact
        </h1>
        <p
          class="italic text-white text-sm ml-32"
        >
          Culture & Art Blog
        </p>
      </div>
      <div
        class="bg-black"
      >
        <div
          class="flex flex-row items-center justify-end mr-20 space-x-10"
        >
          <a
            class="italic text-sm text-white"
            href=""
          >
            Blog
          </a>
          <a
            class="italic text-sm text-white"
            href=""
          >
            About us
          </a>
          <a
            class="italic text-sm text-white"
            href=""
          >
            Contact
          </a>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row flex-wrap gap-x-24 ml-32"
    >
      <div
        class="max-w-xs flex flex-col border-solid border-2 my-3"
        data-testid="card"
      >
        <img
          alt=""
          class="w-full h-3/6"
          src="abstract.png"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <p
            class="pl-1 text-grey"
          >
            2nd Januray, 2018
          </p>
          <p
            class="pr-1 text-grey"
          >
            2 mins
          </p>
        </div>
        <p
          class="text-xl font-bold bold px-3"
        >
          The future of abstract art and the culture ...
        </p>
        <p
          class="text-lg px-3"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </p>
        <hr
          class="h-px w-11/12 mx-auto my-2 bg-gray-200 border-0 dark:bg-gray-700 px-3"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <div
            class="flex flex-row justify-around"
          >
            <button
              data-testid="clap-0"
            >
              <img
                alt="img"
                class=" w-12 px-3"
                src="clapping.svg"
              />
            </button>
            <p>
              10
            </p>
          </div>
          <button>
            <img
              alt="img"
              class="w-12 px-3"
              src="heart-black.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="max-w-xs flex flex-col border-solid border-2 my-3"
        data-testid="card"
      >
        <img
          alt=""
          class="w-full h-3/6"
          src="outdoor.png"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <p
            class="pl-1 text-grey"
          >
            31st Januray, 2018
          </p>
          <p
            class="pr-1 text-grey"
          >
            1 min
          </p>
        </div>
        <p
          class="text-xl font-bold bold px-3"
        >
          How painting outdoors can transform your 
        </p>
        <p
          class="text-lg px-3"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </p>
        <hr
          class="h-px w-11/12 mx-auto my-2 bg-gray-200 border-0 dark:bg-gray-700 px-3"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <div
            class="flex flex-row justify-around"
          >
            <button
              data-testid="clap-1"
            >
              <img
                alt="img"
                class=" w-12 px-3"
                src="clapping.svg"
              />
            </button>
            <p>
              1
            </p>
          </div>
          <button>
            <img
              alt="img"
              class="w-12 px-3"
              src="heart-red.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="max-w-xs flex flex-col border-solid border-2 my-3"
        data-testid="card"
      >
        <img
          alt=""
          class="w-full h-3/6"
          src="young-painters.png"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <p
            class="pl-1 text-grey"
          >
            20th March, 2018
          </p>
          <p
            class="pr-1 text-grey"
          >
            2 mins
          </p>
        </div>
        <p
          class="text-xl font-bold bold px-3"
        >
          10 young painters you need to know
        </p>
        <p
          class="text-lg px-3"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </p>
        <hr
          class="h-px w-11/12 mx-auto my-2 bg-gray-200 border-0 dark:bg-gray-700 px-3"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <div
            class="flex flex-row justify-around"
          >
            <button
              data-testid="clap-2"
            >
              <img
                alt="img"
                class=" w-12 px-3"
                src="clapping.svg"
              />
            </button>
            <p>
              15
            </p>
          </div>
          <button>
            <img
              alt="img"
              class="w-12 px-3"
              src="heart-black.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="max-w-xs flex flex-col border-solid border-2 my-3"
        data-testid="card"
      >
        <img
          alt=""
          class="w-full h-3/6"
          src="perfect-art.png"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <p
            class="pl-1 text-grey"
          >
            22nd June, 2018
          </p>
          <p
            class="pr-1 text-grey"
          >
            5 mins
          </p>
        </div>
        <p
          class="text-xl font-bold bold px-3"
        >
          Tips for creating the perfect art
        </p>
        <p
          class="text-lg px-3"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </p>
        <hr
          class="h-px w-11/12 mx-auto my-2 bg-gray-200 border-0 dark:bg-gray-700 px-3"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <div
            class="flex flex-row justify-around"
          >
            <button
              data-testid="clap-3"
            >
              <img
                alt="img"
                class=" w-12 px-3"
                src="clapping.svg"
              />
            </button>
            <p>
              11
            </p>
          </div>
          <button>
            <img
              alt="img"
              class="w-12 px-3"
              src="heart-black.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="max-w-xs flex flex-col border-solid border-2 my-3"
        data-testid="card"
      >
        <img
          alt=""
          class="w-full h-3/6"
          src="inspired.png"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <p
            class="pl-1 text-grey"
          >
            4th Feburary, 2019
          </p>
          <p
            class="pr-1 text-grey"
          >
            9 mins
          </p>
        </div>
        <p
          class="text-xl font-bold bold px-3"
        >
          Stay inspired: top artists blog ...
        </p>
        <p
          class="text-lg px-3"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </p>
        <hr
          class="h-px w-11/12 mx-auto my-2 bg-gray-200 border-0 dark:bg-gray-700 px-3"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <div
            class="flex flex-row justify-around"
          >
            <button
              data-testid="clap-4"
            >
              <img
                alt="img"
                class=" w-12 px-3"
                src="clapping.svg"
              />
            </button>
            <p>
              45
            </p>
          </div>
          <button>
            <img
              alt="img"
              class="w-12 px-3"
              src="heart-red.svg"
            />
          </button>
        </div>
      </div>
      <div
        class="max-w-xs flex flex-col border-solid border-2 my-3"
        data-testid="card"
      >
        <img
          alt=""
          class="w-full h-3/6"
          src="reason.png"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <p
            class="pl-1 text-grey"
          >
            10th Januray, 2019
          </p>
          <p
            class="pr-1 text-grey"
          >
            2 mins
          </p>
        </div>
        <p
          class="text-xl font-bold bold px-3"
        >
          5 reasons why everyone should try an...
        </p>
        <p
          class="text-lg px-3"
        >
          Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
        </p>
        <hr
          class="h-px w-11/12 mx-auto my-2 bg-gray-200 border-0 dark:bg-gray-700 px-3"
        />
        <div
          class="flex flex-row justify-between align-center"
        >
          <div
            class="flex flex-row justify-around"
          >
            <button
              data-testid="clap-5"
            >
              <img
                alt="img"
                class=" w-12 px-3"
                src="clapping.svg"
              />
            </button>
            <p>
              11
            </p>
          </div>
          <button>
            <img
              alt="img"
              class="w-12 px-3"
              src="heart-black.svg"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex flex-row justify-start items-center bg-black h-24"
    >
      <p
        class="ml-32 text-white"
      >
        @artifact 2019
      </p>
    </div>
  </div>
</DocumentFragment>
`;
